datafusion/core/src/datasource/physical_plan/parquet/arrow_statistics.rs

```diff
@@ -0,0 +1,252 @@
+    /// min values
+    min: ArrayRef,
+    /// max values
+    max: ArrayRef,
+    /// Row counts (UInt64Array)
+    row_count: ArrayRef,
+    /// Null Counts (UInt64Array)
+    null_count: ArrayRef,
+}
+
+impl ArrowStatistics {
+    /// Create a new instance of `ArrowStatistics`
+    pub fn new(
+        min: ArrayRef,
+        max: ArrayRef,
+        row_count: ArrayRef,
+        null_count: ArrayRef,
+    ) -> Self {
+        Self {
+            min,
+            max,
+            row_count,
+            null_count,
+        }
+    }
+
+    /// Get the min values
+    pub fn min(&self) -> &ArrayRef {
+        &self.min
+    }
+
+    /// Get the max values
+    pub fn max(&self) -> &ArrayRef {
+        &self.max
+    }
+
+    /// Get the row counts
+    pub fn row_count(&self) -> &ArrayRef {
+        &self.row_count
+    }
+
+    /// Get the null counts
+    pub fn null_count(&self) -> &ArrayRef {
+        &self.null_count
+    }
+}
+
+/// Extract `ArrowStatistics` from the  parquet [`Statistics`]
+pub fn parquet_stats_to_arrow(
+    arrow_datatype: &DataType,
+    statistics: &ParquetColumnStatistics,
+) -> Result<ArrowStatistics> {
+    // check of the data type is Int64
+    if !matches!(arrow_datatype, DataType::Int64) {
+        return Err(DataFusionError::Internal(format!(
+            "Unsupported data type {:?} for statistics",
+            arrow_datatype
+        )));
+    }
+
+    // row counts
+    let row_count = statistics
+        .rg_statistics
+        .iter()
+        .map(|rg| rg.row_count)
+        .collect::<Vec<_>>();
+
+    // get null counts
+    let parquet_stats = statistics.rg_statistics.iter().map(|rg| rg.statistics);
+    let null_counts = parquet_stats
+        // .map(|stats| stats.and_then(|s| Some(s.null_count())))
+        .map(|stats| stats.map(|s| s.null_count()))
+        .collect::<Vec<_>>();
+
+    // get min and max values
+    let parquet_stats_with_min_max = statistics
+        .rg_statistics
+        .iter()
+        .map(|rg| rg.get_statistics())
+        .collect::<Vec<_>>();
+
+    let mins = parquet_stats_with_min_max
+        .iter()
+        .map(|stats| {
+            stats.and_then(|s| {
+                let stats = ParquetColumnRowGroupStatistics::try_as_i64(s)?;
+                Some(*stats.min())
+            })
+        })
+        .collect::<Vec<_>>();
+
+    let maxs = parquet_stats_with_min_max
+        .iter()
+        .map(|stats| {
+            stats.and_then(|s| {
+                let stats = ParquetColumnRowGroupStatistics::try_as_i64(s)?;
+                Some(*stats.max())
+            })
+        })
+        .collect::<Vec<_>>();
+
+    Ok(ArrowStatistics {
+        min: Arc::new(Int64Array::from(mins)),
+        max: Arc::new(Int64Array::from(maxs)),
+        row_count: Arc::new(UInt64Array::from(row_count)),
+        null_count: Arc::new(UInt64Array::from(null_counts)),
+    })
+}
+
+/// All row group statistics of a file for a column
+pub struct ParquetColumnStatistics<'a> {
```

Comment:
I think this struct will be useful as intermediate steps to get only needed stats from parquet file